File: 3-leetcode/01-easy/js/1957-delete-characters-to-make-fancy-string.ts
const makeFancyString = (s: string): string => {
  if (s.length < 3) {
    return s;
  }

  let res: string = s.substring(0, 2);

  for (let i = 2; i < s.length; i++) {
    if (s[i] !== s[i - 1] || s[i] !== s[i - 2]) {
      res += s[i];
    }
  }

  return res;
};

File: 3-leetcode/01-easy/dart/1957-delete-characters-to-make-fancy-string.dart
class Solution {
  String makeFancyString(String s) {
    if (s.length < 3) {
      return s;
    }

    StringBuffer res = StringBuffer();
    res.write(s[0]);
    res.write(s[1]);

    for (int i = 2; i < s.length; i++) {
      if (s[i] != s[i - 1] || s[i] != s[i - 2]) {
        res.write(s[i]);
      }
    }

    return res.toString();
  }
}
